src/ipc/handler.c

```diff
@@ -809,6 +811,200 @@ static int ipc_glb_gdb_debug(uint32_t header)
 
 }
 
+#if CONFIG_PROBE
+static inline int ipc_probe_init(uint32_t header)
+{
+	struct sof_ipc_probe_dma_add_params *params = _ipc->comp_data;
+	int dma_provided = (params->hdr.size - sizeof(*params)) /
+			   sizeof(struct probe_dma);
```

Comment:
Lets make this explicit in the data structure rather than abstract i.e. have a member that contains this data directly.

Reply:
agree. Having an implicit count is not something we've ever done. we should also add reserved fields, even if it's a debug feature we should not require a new firmware if the kernel is updated and we do want some level of backwards compatibility

Reply:
@lgirdwood @plbossart Again, first and most important requirement that was stated, when I first started working on this feature, was to take cAVS interface directly.
I'm not saying, that your suggestions are invalid, they're totally reasonable and I could happily implement these changes, but then we would have to review the requirements that were set up before this feature was scheduled, and I'm not in power to decide that.
Let us @lbetlej and @mmaka1 chime in on this.

Reply:
If we want to wait for review from so many ppl then it's not going to be merged in this year.

Reply:
ok, so this is to support an existing userspace API which means we need to see how we can work with it today and also keep things easily upgradable for future ABIs.
I'm thinking it may be better to put these "assumptions" as macros or static inlines in a header somewhere that can be updated depending on future ABI updates. They can also be marked with comments so that anyone updating ABI is aware of these assumptions.

Reply:
I've wrapped array size getter into macro `probe_get_payload_array_size` in ipc/handler.c, to encapsulate sizeof arithmetic. I hope that suffices as means of keeping it easy to maintain in case of future ABI changes.